internal/controller/entra/main.go

```diff
@@ -184,3 +185,44 @@ func (c *Controller) AddtoInvitedUserGroup(ctx context.Context, email string) er
 	return err
 
 }
+
+func (c *Controller) SearchForUser(ctx context.Context, query string) ([]string, error) {
+	// do filter query and check in email, principal name and display name
+
+	filterQuery := fmt.Sprintf(
+		`startswith(displayName,'%s') or startswith(userPrincipalName,'%s') or startswith(givenName,'%s') or startswith(mail,'%s')`,
+		query, query, query, query,
```

Comment:
I know it's not sql but this makes me scared about injection. I think the query should be validated to match something like `^\w[\w.\s0-9@]+\w$` (suggestions welcome)

Reply:
presumably earlier is better for validation? so would do it in `GetUsers`?

Reply:
I'd actually do it in this function – so all consumers get a `InvalidObjectError` if the query is invalid